src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs

```diff
@@ -191,224 +204,233 @@ private void CreateStreamAndStartWatcher()
                 _watcherThread.Start();
                 _runLoopStartedEvent.WaitOne();
```

Comment:
Ah cool, good to know...I'll add the Background and pass the ManualResetEventSlim as a parameter